src/Exception/AbstractClassException.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Exception;
+
+/**
+ * @author Morgan Abraham <morgan@geekimo.me>
+ */
+final class AbstractClassException extends \RuntimeException
```

Comment:
```suggestion
final class AbstractClassException extends \InvalidArgumentException
```
Since the argument passed is invalid, what about extending `InvalidArgumentException`?

Reply:
Previously we were throwing a RuntimeException, so extending an InvalidArgumentException would be a BC-break (in case someone was catching RuntimeException), isn't it ?

I think the best we can do is
```
/**
 * NEXT_MAJOR: extends \InvalidArgumentException
 */
final class AbstractClassException extends \RuntimeException
```

Reply:
Done (NEXT_MAJOR comment)

Reply:
It shouldn't be a BC break since`Instantiator` was created as `@internal` IIRC

Reply:
Maybe we can change it before release, then?

Reply:
https://github.com/sonata-project/SonataAdminBundle/pull/7825